WorkflowDemos/WorkflowDemosTest/src_test/test/bpm/TestProcurementRequest.java

```diff
@@ -0,0 +1,225 @@
+   {
+     ui.mock(new PID("15254DC87A1B183B-f3"), in -> testData);
+     List<ITask> verifyTasks = createProcurrementRequest(bpmClient);
+
+     ui.mock(new PID("15254DC87A1B183B-f5"), ProcurementRequest.class, (in, out) -> out.setDataOkTeamLeader(true));
+     verifyRequest(bpmClient, verifyTasks.get(0), "Teamleader");
+
+     ui.mock(new PID("15254DC87A1B183B-f6"), ProcurementRequest.class, (in, out) -> out.setDataOkManager(true));
+     verifyRequest(bpmClient, verifyTasks.get(1), "Manager");
+
+     ICase caze = verifyTasks.get(0).getCase();
+     waitUntilAcceptRequestTaskIsAvailable(caze);
+     
+     ui.mock(new PID("15254DC87A1B183B-f16"), ProcurementRequest.class, (in, out) -> out.setAccepted(true));
+     ProcurementRequest request = acceptRequest(bpmClient, getAcceptRequestTask(caze));
+     
+     assertThat(request.getAccepted()).isTrue();
+     assertThat(caze.getState()).isEqualTo(CaseState.DONE);
+   }
+
+  private List<ITask> createProcurrementRequest(BpmClient bpmClient) throws Exception
+  {
+    ExecutionResult result = bpmClient
+        .start().process(Process.ProcurementRequest+"/start.ivp")
+        .as().user("ldv")
+        .execute();
+    assertThat(result).isNotNull();
+    List<ITask> tasks = result.getRequestedCase().getActiveTasks();
+    assertThat(tasks).hasSize(2);
+    ITask verifyTask1 = tasks.get(0);
+    assertThat(verifyTask1.getName()).startsWith("Pr�fe Antrag:");
+    assertThat(verifyTask1.getActivatorName()).isEqualTo((Object)"Teamleader");
+    ITask verifyTask2 = tasks.get(1);
+    assertThat(verifyTask2.getName()).startsWith("Pr�fe Antrag:");
+    assertThat(verifyTask2.getActivatorName()).isEqualTo((Object)"Manager");
+    return tasks;
+  }
+
+  private void verifyRequest(BpmClient bpmClient, ITask verifyTask, String role) throws Exception
+  {     
+    ExecutionResult result = bpmClient
+        .start().resumableTask(verifyTask)
+        .as().role(role)	
+        .execute();
+    assertThat(result.getRequestedTask().getState()).isIn(TaskState.DONE, TaskState.READY_FOR_JOIN);
+  }
+  
+  private ProcurementRequest acceptRequest(BpmClient bpmClient, ITask acceptRequestTask) throws Exception
+  {
+    ExecutionResult result = bpmClient.start().resumableTask(acceptRequestTask).as().role("Executive Manager").execute();
+    assertThat(result.getRequestedTask().getState()).isIn(TaskState.DONE);
+    return result.getData().last();
+  }
+  
+  //TODO API Sugar: set locale to default cms language. Provide api to set other language
+  private ITask getAcceptRequestTask(ICase caze)
+  {
+    return caze
+        .getActiveTasks()
+        .stream()
+        .filter(task -> task.getName().contains("Antrag annehmen:"))  
+        .findAny()
+        .orElse(null);
+  }
+  
+  // TODO API Sugar: Provide method to wait until system task has been executed 
+  private void waitUntilSystemTaskHasBeenExecuted(ICase caze)
```

Comment:
I'd even vote to enforce systems task to be executed right after a bpm client request has finished ... so users do not even have to trigger any api.

Reply:
Or system tasks are not automatically executed at all and have to be executed by the tester using
as().systemUser()
by doing so he can assert stuff that is done by the system task as well